From 76e0a52f875f9d4d569eee05fe510084d66d992c Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Thu, 7 Aug 2014 18:58:37 +0800
Subject: [PATCH 4751/5965] PD #93541:Remove Wdeclaration-after-statement type
 warning

Change-Id: I0c7e84bb82f5de8b34a65597c57de53d52229c3f

Conflicts:

	sound/soc/aml/m8/aml_m8.c

Conflicts:

	drivers/amlogic/amports/vh265.c

Conflicts:

	sound/soc/aml/m8/aml_spdif_dai.c

Conflicts:

	drivers/amlogic/amports/vh265.c
	sound/soc/aml/m8/aml_spdif_dai.c
---
 Makefile                                      |  3 +-
 arch/arm/mach-meson6/clock.c                  |  2 +-
 drivers/amlogic/amports/vh264_4k2k.c          |  2 +-
 drivers/amlogic/amports/vh265.c               |  2 -
 drivers/amlogic/amports/vreal.c               |  2 +-
 drivers/amlogic/audiodsp/dsp_mailbox.c        |  4 +-
 drivers/amlogic/camera/bf3720.c               |  2 +-
 drivers/amlogic/camera/gc0308.c               |  2 +-
 drivers/amlogic/camera/hm5065.c               |  2 +-
 drivers/amlogic/camera/ov5647.c               | 13 +++---
 drivers/amlogic/deinterlace/deinterlace.c     |  2 +-
 drivers/amlogic/display/osd/osd_antiflicker.c |  2 +-
 drivers/amlogic/display/osd/osd_main.c        |  2 +-
 drivers/amlogic/ethernet/am_net8218.c         | 42 +++++++++++--------
 drivers/amlogic/ethernet/phy/am_ip101.c       |  4 +-
 drivers/amlogic/ethernet/phy/am_micrel.c      |  2 +-
 drivers/amlogic/i2c/aml_i2c.c                 |  2 +-
 .../amlogic/input/new_remote/remote_main.c    |  6 +--
 drivers/amlogic/input/remote/sw_remote_kbd.c  |  6 +--
 drivers/amlogic/input/sensor/lsm303d.c        |  2 +-
 .../touchscreen/gslx680/amlogic_gslX680.c     | 12 +++---
 drivers/amlogic/ion_dev/dev_ion.c             |  2 +-
 .../meson_cs_dcdc_regulator.c                 | 14 +++----
 drivers/amlogic/spi_nor/spi_nor.c             | 18 ++++----
 drivers/amlogic/wifi/wifi_power.c             |  2 +-
 drivers/media/dvb-core/dvb_frontend.c         |  4 +-
 drivers/staging/android/lowmemorykiller.c     |  4 +-
 fs/ext4/super.c                               |  2 +-
 kernel/power/suspend.c                        |  3 +-
 sound/soc/aml/m6/aml_pcm.c                    |  3 +-
 sound/soc/aml/m6/aml_platform.c               |  6 +--
 sound/soc/aml/m8/aml_i2s.c                    |  2 +-
 sound/soc/aml/m8/aml_m8.c                     |  3 +-
 sound/soc/aml/m8/aml_spdif_codec.c            |  2 +-
 sound/soc/aml/m8/aml_spdif_dai.c              |  4 +-
 sound/soc/codecs/aml_m8_codec.c               |  2 +-
 sound/soc/codecs/wm8960.c                     |  2 +-
 37 files changed, 101 insertions(+), 88 deletions(-)

diff --git a/Makefile b/Makefile
index 9d845152b894..75eede736640 100755
--- a/Makefile
+++ b/Makefile
@@ -384,7 +384,8 @@ KBUILD_CFLAGS   += -Werror=enum-compare \
 		   -Werror=format \
 		   -Werror=unused-function \
 		   -Werror=switch \
-		   -Werror=strict-prototypes
+		   -Werror=strict-prototypes \
+		   -Werror=declaration-after-statement
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
diff --git a/arch/arm/mach-meson6/clock.c b/arch/arm/mach-meson6/clock.c
index 740a4cfc6741..25fe05085cd4 100755
--- a/arch/arm/mach-meson6/clock.c
+++ b/arch/arm/mach-meson6/clock.c
@@ -1418,10 +1418,10 @@ static int clk_set_rate_a9(struct clk *clk, unsigned long rate)
 #endif /* CONFIG_SMP */
 	int error = 0;
 
+	int cpu_freq_limit = 1200000000;
 	if (rate < 1000)
 		rate *= 1000000;
 
-	int cpu_freq_limit = 1200000000;
 	if(device_low_power != 0) {
 		cpu_freq_limit = 600000000;
 	}
diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 4153afd2cc18..b19d61676741 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -1405,9 +1405,9 @@ static int amvdec_h264_4k2k_probe(struct platform_device *pdev)
 
     if (!H264_4K2K_SINGLE_CORE) {
 #if (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8)&&(HAS_HDEC)
+        int count = 0;
         if(get_vdec2_usage() != USAGE_NONE)
             AbortEncodeWithVdec2(1);
-        int count = 0;
         while((get_vdec2_usage() != USAGE_NONE)&&(count < 10)){
             msleep(50);
             count++;
diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index 65943d3d49fd..98cb963d7cd0 100755
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -3061,8 +3061,6 @@ static void hevc_recover(hevc_stru_t* hevc)
         amhevc_stop();
 
         //reset
-        WRITE_MPEG_REG(PARSER_VIDEO_RP, READ_VREG(HEVC_STREAM_RD_PTR));
-        SET_MPEG_REG_MASK(PARSER_ES_CONTROL, ES_VID_MAN_RD_PTR);
 
         hevc_stream_start_addr = READ_VREG(HEVC_STREAM_START_ADDR);
         hevc_stream_end_addr = READ_VREG(HEVC_STREAM_END_ADDR);
diff --git a/drivers/amlogic/amports/vreal.c b/drivers/amlogic/amports/vreal.c
index 0ad911ece4ca..1e3159551877 100755
--- a/drivers/amlogic/amports/vreal.c
+++ b/drivers/amlogic/amports/vreal.c
@@ -828,9 +828,9 @@ static int amvdec_real_probe(struct platform_device *pdev)
 #if (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8)&&(HAS_HDEC)
     if(IS_MESON_M8_CPU){
         // disable vdec2 dblk when miracast.
+        int count = 0;
         if(get_vdec2_usage() != USAGE_NONE)
             AbortEncodeWithVdec2(1);
-        int count = 0;
         while((get_vdec2_usage() != USAGE_NONE)&&(count < 10)){
             msleep(50);
             count++;
diff --git a/drivers/amlogic/audiodsp/dsp_mailbox.c b/drivers/amlogic/audiodsp/dsp_mailbox.c
index f0c3e164503d..2a665a5fb88e 100755
--- a/drivers/amlogic/audiodsp/dsp_mailbox.c
+++ b/drivers/amlogic/audiodsp/dsp_mailbox.c
@@ -110,9 +110,11 @@ static irqreturn_t audiodsp_mailbox_irq(int irq, void *data)
 	unsigned long status;
 	struct mail_msg msg;
 	int i = 0;
-	status=READ_VREG(MB1_REG);
 #if MESON_CPU_TYPE < MESON_CPU_TYPE_MESON8	
 	unsigned long fiq_mask;
+#endif
+	status=READ_VREG(MB1_REG);
+#if MESON_CPU_TYPE < MESON_CPU_TYPE_MESON8	
 	fiq_mask=READ_VREG(MB1_SEL);
 	status=status&fiq_mask;
 #endif
diff --git a/drivers/amlogic/camera/bf3720.c b/drivers/amlogic/camera/bf3720.c
index b851d3f7febd..6ac2c2225eb2 100755
--- a/drivers/amlogic/camera/bf3720.c
+++ b/drivers/amlogic/camera/bf3720.c
@@ -2072,13 +2072,13 @@ static int vidioc_s_fmt_vid_cap(struct file *file, void *priv,
 	struct bf3720_fh *fh = priv;
 	struct videobuf_queue *q = &fh->vb_vidq;
 	struct bf3720_device *dev = fh->dev;
+	int ret = vidioc_try_fmt_vid_cap(file, fh, f);
 
         f->fmt.pix.width = (f->fmt.pix.width + (CANVAS_WIDTH_ALIGN-1) ) & (~(CANVAS_WIDTH_ALIGN-1));
 	if ((f->fmt.pix.pixelformat==V4L2_PIX_FMT_YVU420) ||
             (f->fmt.pix.pixelformat==V4L2_PIX_FMT_YUV420)){
                 f->fmt.pix.width = (f->fmt.pix.width + (CANVAS_WIDTH_ALIGN*2-1) ) & (~(CANVAS_WIDTH_ALIGN*2-1));
         }
-	int ret = vidioc_try_fmt_vid_cap(file, fh, f);
 	if (ret < 0)
 		return ret;
 
diff --git a/drivers/amlogic/camera/gc0308.c b/drivers/amlogic/camera/gc0308.c
index 966ef3e6e49e..d10e210da76c 100755
--- a/drivers/amlogic/camera/gc0308.c
+++ b/drivers/amlogic/camera/gc0308.c
@@ -2030,8 +2030,8 @@ buffer_setup(struct videobuf_queue *vq, unsigned int *count, unsigned int *size)
 	struct gc0308_fh *fh = container_of(res, struct gc0308_fh, res);
 	struct gc0308_device *dev  = fh->dev;
     //int bytes = fh->fmt->depth >> 3 ;
-	*size = fh->width*fh->height*fh->fmt->depth >> 3;
 	int height = fh->height;
+	*size = fh->width*fh->height*fh->fmt->depth >> 3;
 	if(height==1080)
 		height = 1088;
 	*size = (fh->width*height*fh->fmt->depth)>>3;
diff --git a/drivers/amlogic/camera/hm5065.c b/drivers/amlogic/camera/hm5065.c
index 0d527ef44968..61396f924653 100755
--- a/drivers/amlogic/camera/hm5065.c
+++ b/drivers/amlogic/camera/hm5065.c
@@ -1845,12 +1845,12 @@ static int vidioc_s_fmt_vid_cap(struct file *file, void *priv,
 	//unsigned char gain = 0, exposurelow = 0, exposuremid = 0, exposurehigh = 0;
 	int cap_fps, pre_fps;
 
+	int ret = vidioc_try_fmt_vid_cap(file, fh, f);
         f->fmt.pix.width = (f->fmt.pix.width + (CANVAS_WIDTH_ALIGN-1) ) & (~(CANVAS_WIDTH_ALIGN-1));
 	if ((f->fmt.pix.pixelformat==V4L2_PIX_FMT_YVU420) ||
             (f->fmt.pix.pixelformat==V4L2_PIX_FMT_YUV420)){
                 f->fmt.pix.width = (f->fmt.pix.width + (CANVAS_WIDTH_ALIGN*2-1) ) & (~(CANVAS_WIDTH_ALIGN*2-1));
         }
-	int ret = vidioc_try_fmt_vid_cap(file, fh, f);
 	if (ret < 0)
     	return ret;
 
diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 38a0cbee25de..9e3cf31faeae 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -2365,9 +2365,9 @@ bool OV5647_set_af_new_step(void *priv, unsigned int af_step){
 
 void OV5647_set_new_format(void *priv,int width,int height,int fr){
     int index = 0;
-    current_fr = fr;
     camera_priv_data_t *camera_priv_data = (camera_priv_data_t *)priv;
     configure_t *configure = camera_priv_data->configure;
+    current_fr = fr;
     if(camera_priv_data == NULL)
     	return;
     printk("sum:%d,mode:%d,fr:%d\n",configure->aet.sum,ov5647_work_mode,fr);
@@ -3062,6 +3062,12 @@ void set_resolution_param(struct ov5647_device *dev, resolution_param_t* res_par
     //int rc = -1;
     int i=0;
     unsigned char t = dev->cam_info.interface;
+	int default_sensor_data[4] = {0x00000668,0x00000400,0x00000400,0x00000878};
+    int *sensor_data;
+    int addr_start = 0x5186;
+    int data = 0;
+    int index = 0;
+
     if(i_index != -1 && ov5647_work_mode != CAMERA_CAPTURE){
     	printk("i_index is %d\n", i_index);
         res_param = &debug_prev_resolution_array[i_index];
@@ -3086,11 +3092,6 @@ void set_resolution_param(struct ov5647_device *dev, resolution_param_t* res_par
     
     set_flip(dev);
 
-    int default_sensor_data[4] = {0x00000668,0x00000400,0x00000400,0x00000878};
-    int *sensor_data;
-    int addr_start = 0x5186;
-    int data = 0;
-    int index = 0;
     if(dev->configure->wb_sensor_data_valid == 1){
         sensor_data = dev->configure->wb_sensor_data.export;
     }else
diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 1fc5158ac9a6..557756df6c68 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -5521,6 +5521,7 @@ static int process_post_vframe(void)
     di_buf_t *p = NULL;//, *ptmp;
     int itmp;
     int ready_count = list_count(QUEUE_PRE_READY);
+    bool check_drop = false;
     if(queue_empty(QUEUE_POST_FREE)){
         return 0;
     }
@@ -5795,7 +5796,6 @@ static int process_post_vframe(void)
                     }
 #endif
                     di_lock_irqfiq_save(irq_flag2, fiq_flag);
-                    bool check_drop = false;
                     if((check_start_drop_prog && is_progressive(ready_di_buf->vframe))||
                     	  !is_progressive(ready_di_buf->vframe))
                         check_drop = true;
diff --git a/drivers/amlogic/display/osd/osd_antiflicker.c b/drivers/amlogic/display/osd/osd_antiflicker.c
index fc74604276dc..cb2617dd05aa 100755
--- a/drivers/amlogic/display/osd/osd_antiflicker.c
+++ b/drivers/amlogic/display/osd/osd_antiflicker.c
@@ -201,9 +201,9 @@ static int osd_antiflicker_process(void)
 	u32 y1 = 0;
 	u32 yres = 0;
 
-	mutex_lock(&osd_antiflicker_mutex);
 	config_para_ex_t *ge2d_config = &ge2d_osd_antiflicker.ge2d_config;
 	ge2d_context_t *context = ge2d_osd_antiflicker.ge2d_context;
+	mutex_lock(&osd_antiflicker_mutex);
 
 	canvas_read(OSD1_CANVAS_INDEX, &cs);
 	canvas_read(OSD1_CANVAS_INDEX, &cd);
diff --git a/drivers/amlogic/display/osd/osd_main.c b/drivers/amlogic/display/osd/osd_main.c
index 041c44e31c71..63ed94980111 100755
--- a/drivers/amlogic/display/osd/osd_main.c
+++ b/drivers/amlogic/display/osd/osd_main.c
@@ -1730,8 +1730,8 @@ osd_probe(struct platform_device *pdev)
 		mydef_var[index].height=vinfo->screen_real_height;
 		if( init_logo_obj && index==logo_osd_index ) //adjust default var info
 		{
-			printk("don't find to display_size_default from mesonfb-dts\n");
 			int  bpp=init_logo_obj->dev->output_dev.osd.color_depth;//bytes per pixel
+			printk("don't find to display_size_default from mesonfb-dts\n");
 			mydef_var[index].xres=init_logo_obj->dev->vinfo->width;
 			mydef_var[index].yres=init_logo_obj->dev->vinfo->height;	
 			mydef_var[index].xres_virtual=init_logo_obj->dev->vinfo->width;
diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 52d93aafcea0..e9536d377530 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -667,15 +667,15 @@ void net_tasklet(unsigned long dev_instance)
 	}
 	if (result & 2) {
 		struct _rx_desc *c_rx, *rx = NULL;
+		int rx_cnt = 0;
 		c_rx = (void *)readl((void*)(np->base_addr + ETH_DMA_19_Curr_Host_Re_Descriptor));
 		c_rx = np->rx_ring + (c_rx - np->rx_ring_dma);
 		rx = np->last_rx->next;
-		int rx_cnt = 0;
 		while (rx != NULL) {
 			CACHE_RSYNC(rx, sizeof(struct _rx_desc));
 			if (!(rx->status & (DescOwnByDma))) {
-				rx_cnt++;
 				int ip_summed = CHECKSUM_UNNECESSARY;
+				rx_cnt++;
 				len = (rx->status & DescFrameLengthMask) >> DescFrameLengthShift;
 				if (unlikely(len < 18 || len > np->rx_buf_sz)) {	//here is fatal error we drop it ;
 					np->stats.rx_dropped++;
@@ -895,24 +895,14 @@ static int mac_pmt_enable(unsigned int enable)
 extern int get_aml_key_kernel(const char* key_name, unsigned char* data, int ascii_flag);
 extern int extenal_api_key_set_version(char *devvesion);
 static char print_buff[1025];
-#endif
-static int aml_mac_init(struct net_device *ndev)
+void read_mac_from_nand(struct net_device *ndev)
 {
-	struct am_net_private *np = netdev_priv(ndev);
-	unsigned long val;
-
-	writel(1, (void*)(np->base_addr + ETH_DMA_0_Bus_Mode));
-	writel(0x00100800,(void*)(np->base_addr + ETH_DMA_0_Bus_Mode));
-	printk("--1--write mac add to:");
-
-	data_dump(ndev->dev_addr, 6);
-#ifdef CONFIG_AML_NAND_KEY
 	int ret;
 	u8 mac[ETH_ALEN];
 	char *endp;
 	int j;
-	extenal_api_key_set_version("nand3");
 	ret = get_aml_key_kernel("mac", print_buff, 0);
+	extenal_api_key_set_version("nand3");
 	printk("ret = %d\nprint_buff=%s\n", ret, print_buff);
 	if (ret >= 0) {
 		strcpy(ndev->dev_addr, print_buff);
@@ -923,6 +913,21 @@ static int aml_mac_init(struct net_device *ndev)
 	}
 	memcpy(ndev->dev_addr, mac, ETH_ALEN);
 	}
+
+}
+#endif
+static int aml_mac_init(struct net_device *ndev)
+{
+	struct am_net_private *np = netdev_priv(ndev);
+	unsigned long val;
+
+	writel(1, (void*)(np->base_addr + ETH_DMA_0_Bus_Mode));
+	writel(0x00100800,(void*)(np->base_addr + ETH_DMA_0_Bus_Mode));
+	printk("--1--write mac add to:");
+
+	data_dump(ndev->dev_addr, 6);
+#ifdef CONFIG_AML_NAND_KEY
+	read_mac_from_nand(ndev);
 #endif
 	printk("--2--write mac add to:");
 	data_dump(ndev->dev_addr, 6);
@@ -1225,9 +1230,9 @@ static int netdev_open(struct net_device *dev)
 {
 	struct am_net_private *np = netdev_priv(dev);
 	unsigned long val;
+	int res;
 	np->refcnt++;
 	switch_mod_gate_by_name("ethernet",1);
-	int res;
 	if (running) {
 		return 0;
 	}
@@ -1731,8 +1736,8 @@ static void set_multicast_list(struct net_device *dev)
 	}
 }
 static int set_mac_addr_n(struct net_device *dev, void *addr){
-	printk("mac addr come in\n");
 	struct sockaddr *sa = addr;
+	printk("mac addr come in\n");
 
 	if (!is_valid_ether_addr(sa->sa_data))
 		return -EADDRNOTAVAIL;
@@ -2757,8 +2762,8 @@ static int am_net_cali(int argc, char **argv,int gate)
 	int cali_start = 0;
 	int cali_time = 0;
 	int ii=0;
-	cali_start = gate;
 	unsigned int value;
+	cali_start = gate;
 	if ((argc < 4) || (argv == NULL) || (argv[0] == NULL)
 			|| (argv[1] == NULL) || (argv[2] == NULL)|| (argv[3] == NULL)) {
 		printk("Invalid syntax\n");
@@ -2906,6 +2911,7 @@ static int ethernet_probe(struct platform_device *pdev)
 {
         int ret;
 	int res;
+	struct am_net_private *np=NULL;
 	printk("ethernet_driver probe!\n");
 #ifdef CONFIG_OF
 	if (!pdev->dev.of_node) {
@@ -2973,7 +2979,7 @@ static int ethernet_probe(struct platform_device *pdev)
 		res = am_eth_class_init();
 
 	eth_pdata = (struct aml_eth_platdata *)pdev->dev.platform_data;
-	struct am_net_private *np = netdev_priv(my_ndev);
+	np = netdev_priv(my_ndev);
 	if(np->phydev && savepowermode)
 		np->phydev->drv->suspend(np->phydev);
 	//switch_mod_gate_by_name("ethernet",0);
diff --git a/drivers/amlogic/ethernet/phy/am_ip101.c b/drivers/amlogic/ethernet/phy/am_ip101.c
index 1ccfebb0bf3c..b1a2a27ce52e 100644
--- a/drivers/amlogic/ethernet/phy/am_ip101.c
+++ b/drivers/amlogic/ethernet/phy/am_ip101.c
@@ -264,8 +264,8 @@ static int ip101a_g_ack_interrupt(struct phy_device *phydev)
 }
 static int ip101a_g_genphy_suspend(struct phy_device *phydev)
 {
-	printk("**************ip101 sd*****************\n");
 	int value;
+	printk("**************ip101 sd*****************\n");
 	off_analog(phydev, OFF);
 	value = phy_read(phydev, MII_BMCR);
 	phy_write(phydev, MII_BMCR, (value | BMCR_PDOWN));
@@ -273,8 +273,8 @@ static int ip101a_g_genphy_suspend(struct phy_device *phydev)
 }
 static int ip101a_g_genphy_resume(struct phy_device *phydev)
 {
-	printk("**************ip101 re*****************\n");
 	int value;
+	printk("**************ip101 re*****************\n");
 	off_analog(phydev, ON);
 	value = phy_read(phydev, MII_BMCR);
 	phy_write(phydev, MII_BMCR, (value & ~BMCR_PDOWN));
diff --git a/drivers/amlogic/ethernet/phy/am_micrel.c b/drivers/amlogic/ethernet/phy/am_micrel.c
index 51b91c4a2916..7a55256336c7 100755
--- a/drivers/amlogic/ethernet/phy/am_micrel.c
+++ b/drivers/amlogic/ethernet/phy/am_micrel.c
@@ -254,6 +254,7 @@ static int ksz8091_set_wol(struct phy_device *phydev, struct ethtool_wolinfo *wo
 {
 	int err, oldpage, temp;
 
+	int i = 0;
 	oldpage = phy_read(phydev, KSZ8091_MMD_CTRL);
 /*
 Magic-packet detection is enabled by writing a 1 to MMD address 1Fh,register 0h, bit [6]
@@ -261,7 +262,6 @@ Magic-packet detection is enabled by writing a 1 to MMD address 1Fh,register 0h,
 The KSZ8091MNX/RNB does not generate the magic packet. The magic packet must be provided by the external system.
 */
 	printk("wol->wolopts = %d\nWAKE_MAGIC = %d\n",wol->wolopts,WAKE_MAGIC);
-	int i = 0;
 	if (wol->wolopts & WAKE_MAGIC) {
 	temp =  phy_read(phydev, KSZ8091_WOL_OMSO);
 
diff --git a/drivers/amlogic/i2c/aml_i2c.c b/drivers/amlogic/i2c/aml_i2c.c
index 82c83ce9d72e..eb7d201ade61 100755
--- a/drivers/amlogic/i2c/aml_i2c.c
+++ b/drivers/amlogic/i2c/aml_i2c.c
@@ -1027,6 +1027,7 @@ static int aml_i2c_probe(struct platform_device *pdev)
 	resource_size_t *res_start;
 	struct aml_i2c *i2c = kzalloc(sizeof(struct aml_i2c), GFP_KERNEL);
 
+		struct aml_i2c_platform *aml_i2c_property = kzalloc(sizeof(struct aml_i2c_platform), GFP_KERNEL);
 	//printk(KERN_DEBUG "%s : %s\n", __FILE__, __FUNCTION__);
 
 	if (!pdev->dev.of_node) {
@@ -1042,7 +1043,6 @@ static int aml_i2c_probe(struct platform_device *pdev)
 
 	pdev->id = device_id;
 
-		struct aml_i2c_platform *aml_i2c_property = kzalloc(sizeof(struct aml_i2c_platform), GFP_KERNEL);
 
 		if(!aml_i2c_property)
 			printk("can't alloc mem for i2c_property\n");
diff --git a/drivers/amlogic/input/new_remote/remote_main.c b/drivers/amlogic/input/new_remote/remote_main.c
index 36024daa440e..ec209f11bce9 100755
--- a/drivers/amlogic/input/new_remote/remote_main.c
+++ b/drivers/amlogic/input/new_remote/remote_main.c
@@ -149,8 +149,6 @@ int remote_printk(const char *fmt, ...)
 
 static int remote_mouse_event(struct input_dev *dev, unsigned int scancode, unsigned int type,bool flag)
 {
-	if(flag)
-		return 1;
 	
 	__u16 mouse_code = REL_X;
 	__s32 mouse_value = 0;
@@ -158,6 +156,8 @@ static int remote_mouse_event(struct input_dev *dev, unsigned int scancode, unsi
 	__s32 move_accelerate[] = {0, 2, 2, 4, 4, 6, 8, 10, 12, 14, 16, 18};
 	unsigned int i;
 
+	if(flag)
+		return 1;
 	for (i = 0; i < ARRAY_SIZE(mouse_map[gp_remote->map_num]); i++)
 		if (mouse_map[gp_remote->map_num][i] == scancode) {
 			break;
@@ -627,8 +627,8 @@ static int remote_probe(struct platform_device *pdev)
 	struct remote *remote;
 	struct input_dev *input_dev;
 	unsigned int ao_baseaddr;
-	aml_set_reg32_mask(P_AO_RTI_PIN_MUX_REG, (1 << 0));
 	int i, ret;
+	aml_set_reg32_mask(P_AO_RTI_PIN_MUX_REG, (1 << 0));
 	if (!pdev->dev.of_node) {
 		printk("aml_remote: pdev->dev.of_node == NULL!\n");
 		return -1;
diff --git a/drivers/amlogic/input/remote/sw_remote_kbd.c b/drivers/amlogic/input/remote/sw_remote_kbd.c
index 6abfdbd15533..876fe4289366 100755
--- a/drivers/amlogic/input/remote/sw_remote_kbd.c
+++ b/drivers/amlogic/input/remote/sw_remote_kbd.c
@@ -292,14 +292,14 @@ static inline void kbd_software_mode_remote_data(unsigned long data)
 				remote_data->send_data = 0;
 			}
 			else {
-				remote_data->step = REMOTE_STATUS_WAIT;
 				int valid = checkKeyCode(remote_data->cur_keycode, remote_data->time_window[0]);
+				unsigned int cur_jiffies = jiffies;
+				unsigned int cur_keycode = (remote_data->cur_keycode&0xff00)>>8;
+				remote_data->step = REMOTE_STATUS_WAIT;
 				if(valid || ((remote_data->register_data&0xffff) != remote_data->custom_code)) {
 					input_dbg("***invalid code=%08x-%08x, customer=%08x.\n", remote_data->register_data, remote_data->cur_keycode, remote_data->custom_code);
 					break;
 				}
-				unsigned int cur_jiffies = jiffies;
-				unsigned int cur_keycode = (remote_data->cur_keycode&0xff00)>>8;
 				// press A, press B before A release time out, so we need to send A release
 				if((remote_data->timer.expires > cur_jiffies) && (remote_data->last_keycode != cur_keycode)) {
 					remote_send_key(remote_data->input, remote_data->last_keycode, 0);
diff --git a/drivers/amlogic/input/sensor/lsm303d.c b/drivers/amlogic/input/sensor/lsm303d.c
index 4fe10a1929b3..89e7e206482d 100644
--- a/drivers/amlogic/input/sensor/lsm303d.c
+++ b/drivers/amlogic/input/sensor/lsm303d.c
@@ -2621,11 +2621,11 @@ static int create_sysfs_interfaces(struct lsm303d_status *stat)
 	int i=0;
 //	struct lsm303d_status *stat = dev_get_drvdata(dev);
 
+	struct kobject *mag_kobj = &stat->input_dev_mag->dev.kobj ;
 	struct kobject *acc_kobj = &stat->input_dev_acc->dev.kobj ;
 	if(!acc_kobj)
 		return -ENOMEM;
 
-	struct kobject *mag_kobj = &stat->input_dev_mag->dev.kobj ;
 	if(!mag_kobj)
 		return -ENOMEM;
 
diff --git a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
index 4151c02a4cc3..9bdb9801c188 100755
--- a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
+++ b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
@@ -697,6 +697,12 @@ static void gslX680_ts_worker(struct work_struct *work)
 	u16 x, y;
 
 	struct gsl_ts *ts = container_of(work, struct gsl_ts,work);
+#ifdef GSL_NOID_VERSION
+	u32 tmp1;
+	u8 buf[4] = {0};
+	struct gsl_touch_info cinfo;
+	memset(&cinfo,0,sizeof(struct gsl_touch_info));
+#endif
 
 	print_info("=====gslX680_ts_worker=====\n");				 
 
@@ -707,12 +713,6 @@ static void gslX680_ts_worker(struct work_struct *work)
 		i2c_lock_flag = 1;
 #endif
 
-#ifdef GSL_NOID_VERSION
-	u32 tmp1;
-	u8 buf[4] = {0};
-	struct gsl_touch_info cinfo;
-	memset(&cinfo,0,sizeof(struct gsl_touch_info));
-#endif
 
 	rc = gsl_ts_read(ts->client, 0x80, ts->touch_data, ts->dd->data_size);
 	if (rc < 0) 
diff --git a/drivers/amlogic/ion_dev/dev_ion.c b/drivers/amlogic/ion_dev/dev_ion.c
index 695557196a68..7b345d276de9 100755
--- a/drivers/amlogic/ion_dev/dev_ion.c
+++ b/drivers/amlogic/ion_dev/dev_ion.c
@@ -46,13 +46,13 @@ static struct resource memobj;
 int dev_ion_probe(struct platform_device *pdev) {
     int err;
     int i;
-    num_heaps = 1;
     
     struct resource *res;
     struct device_node	*of_node = pdev->dev.of_node;
     const void *name;
     int offset,size;
 
+    num_heaps = 1;
     my_ion_heap[0].type = ION_HEAP_TYPE_SYSTEM;
     my_ion_heap[0].id = ION_HEAP_TYPE_SYSTEM;
     my_ion_heap[0].name = "vmalloc_ion";
diff --git a/drivers/amlogic/power/meson_cs_dcdc_regulator/meson_cs_dcdc_regulator.c b/drivers/amlogic/power/meson_cs_dcdc_regulator/meson_cs_dcdc_regulator.c
index 6d722a6c7e22..4ab26e49c859 100755
--- a/drivers/amlogic/power/meson_cs_dcdc_regulator/meson_cs_dcdc_regulator.c
+++ b/drivers/amlogic/power/meson_cs_dcdc_regulator/meson_cs_dcdc_regulator.c
@@ -355,8 +355,11 @@ static void vcck_pwm_init(struct device * dev) {
 static 
 int meson_cs_probe(struct platform_device *pdev)
 {
-	
-	printk("======================enter %s done!\n",__func__);
+	struct meson_cs_pdata_t *meson_cs_pdata  = pdev->dev.platform_data;
+	struct meson_cs_regulator_dev *meson_cs_regulator;
+	int error = 0, i, cur_idx;
+	struct regulator_config *meson_regulator_config;
+
 #ifdef CONFIG_USE_OF
 	struct device_node *np = pdev->dev.of_node;
 	struct device_node *np_init_data;
@@ -438,11 +441,8 @@ int meson_cs_probe(struct platform_device *pdev)
 	}
 #endif
 
-	struct meson_cs_pdata_t *meson_cs_pdata  = pdev->dev.platform_data;
-	struct meson_cs_regulator_dev *meson_cs_regulator;
-	int error = 0, i, cur_idx;
-	struct regulator_config *meson_regulator_config;
-       g_vcck_voltage = meson_cs_pdata;
+	printk("======================enter %s done!\n",__func__);
+	       g_vcck_voltage = meson_cs_pdata;
 	meson_cs_regulator = kzalloc(sizeof(struct meson_cs_regulator_dev), GFP_KERNEL);
 	if (!meson_cs_regulator)
 		return -ENOMEM;
diff --git a/drivers/amlogic/spi_nor/spi_nor.c b/drivers/amlogic/spi_nor/spi_nor.c
index bb5aa303938a..59905d502ca8 100755
--- a/drivers/amlogic/spi_nor/spi_nor.c
+++ b/drivers/amlogic/spi_nor/spi_nor.c
@@ -607,15 +607,7 @@ early_param("storage",get_storage_device);
 	 * a chip ID, try the JEDEC id commands; they'll work for most
 	 * newer chips, even if we don't recognize the particular chip.
 	 */
-	printk("%s\n", __func__);
 		
-	int flag = -1;
-	flag = check_storage_device();
-	if(flag < 0){
-		printk("%s %d boot_device_flag %d : do not init spi\n",__func__,__LINE__,boot_device_flag);
-		return  -ENOMEM;
-	}
-	
 #ifdef CONFIG_OF
 	int index;
 	int val = 0;
@@ -625,6 +617,16 @@ early_param("storage",get_storage_device);
 	int ret;
 	phandle phandle;
 	struct device_node *np_spi_part;
+#endif
+	int flag = -1;
+	flag = check_storage_device();
+	printk("%s\n", __func__);
+	if(flag < 0){
+		printk("%s %d boot_device_flag %d : do not init spi\n",__func__,__LINE__,boot_device_flag);
+		return  -ENOMEM;
+	}
+	
+#ifdef CONFIG_OF
 
 	if(spi->dev.of_node){
 		data = kzalloc(sizeof(struct flash_platform_data), GFP_KERNEL);
diff --git a/drivers/amlogic/wifi/wifi_power.c b/drivers/amlogic/wifi/wifi_power.c
index 32a18314ea0a..a68a643ff58f 100755
--- a/drivers/amlogic/wifi/wifi_power.c
+++ b/drivers/amlogic/wifi/wifi_power.c
@@ -256,8 +256,8 @@ EXPORT_SYMBOL(wifi_get_country_code);
 
 int wifi_usb_set_power(int val)
 {
-	WARN_ON(1);
     struct wifi_power_platform_data *pdata = NULL;
+	WARN_ON(1);
     if(devp==NULL)
         return -1;
     pdata = (struct wifi_power_platform_data*)devp->platform_data;
diff --git a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
index cb6c1bd1feae..487f3aaa74ee 100755
--- a/drivers/media/dvb-core/dvb_frontend.c
+++ b/drivers/media/dvb-core/dvb_frontend.c
@@ -2502,11 +2502,11 @@ out:
 
 static int dtv_set_frontend(struct dvb_frontend *fe)
 {
-	printk("dtv_set_frontend\n");
 	struct dvb_frontend_private *fepriv = fe->frontend_priv;
 	struct dtv_frontend_properties *c = &fe->dtv_property_cache;
 	struct dvb_frontend_tune_settings fetunesettings;
 	u32 rolloff = 0;
+	printk("dtv_set_frontend\n");
 
 	if (dvb_frontend_check_parameters(fe) < 0)
 		return -EINVAL;
@@ -3240,6 +3240,7 @@ int dvb_register_frontend(struct dvb_adapter* dvb,
 		.kernel_ioctl = dvb_frontend_ioctl
 	};
 
+	int ret;
 	dev_dbg(dvb->device, "%s:\n", __func__);
 
 	if (mutex_lock_interruptible(&frontend_mutex))
@@ -3277,7 +3278,6 @@ int dvb_register_frontend(struct dvb_adapter* dvb,
 
 	dvb_register_device (fe->dvb, &fepriv->dvbdev, &dvbdev_template,
 			     fe, DVB_DEVICE_FRONTEND);
-	int ret;
 	printk("For tongfang\n");
 	ret=0;
 	tongfang_clsp = class_create(THIS_MODULE,LOCK_DEVICE_NAME);
diff --git a/drivers/staging/android/lowmemorykiller.c b/drivers/staging/android/lowmemorykiller.c
index 84ee8ddae9f2..c0f969e266eb 100644
--- a/drivers/staging/android/lowmemorykiller.c
+++ b/drivers/staging/android/lowmemorykiller.c
@@ -80,12 +80,12 @@ static int lowmem_shrink(struct shrinker *s, struct shrink_control *sc)
 	int other_free = global_page_state(NR_FREE_PAGES) - totalreserve_pages;
 
     /* For request of unmovable pages, take no account of free CMA pages*/
-    if(IS_ENABLED(CONFIG_CMA) && (allocflags_to_migratetype(sc->gfp_mask) != MIGRATE_MOVABLE))
-        other_free -= global_page_state(NR_FREE_CMA_PAGES);
 
 	int other_file = global_page_state(NR_FILE_PAGES) -
 						global_page_state(NR_SHMEM);
 
+    if(IS_ENABLED(CONFIG_CMA) && (allocflags_to_migratetype(sc->gfp_mask) != MIGRATE_MOVABLE))
+        other_free -= global_page_state(NR_FREE_CMA_PAGES);
 	if (lowmem_adj_size < array_size)
 		array_size = lowmem_adj_size;
 	if (lowmem_minfree_size < array_size)
diff --git a/fs/ext4/super.c b/fs/ext4/super.c
index 90e6369fe830..ec884032cd3c 100644
--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -4046,8 +4046,8 @@ no_journal:
 		descr = "out journal";
 
 	if (test_opt(sb, DISCARD)) {
-	    printk("checked enable EXT4 DISCARD here\n");
 		struct request_queue *q = bdev_get_queue(sb->s_bdev);
+	    printk("checked enable EXT4 DISCARD here\n");
 		if (!blk_queue_discard(q))
 			ext4_msg(sb, KERN_WARNING,
 				 "mounting with \"discard\" option, but "
diff --git a/kernel/power/suspend.c b/kernel/power/suspend.c
index d1222cd3eaf3..94655b279e17 100755
--- a/kernel/power/suspend.c
+++ b/kernel/power/suspend.c
@@ -135,7 +135,7 @@ static int suspend_test(int level)
  */
  
 int deep_suspend_flag=0;
-
+extern void clr_pwr_key(void);
 static int suspend_prepare(suspend_state_t state)
 {
 	int error;
@@ -151,7 +151,6 @@ static int suspend_prepare(suspend_state_t state)
 
 	deep_suspend_flag=1;
 	#ifdef CONFIG_AML_GPIO_KEY
-	extern void clr_pwr_key(void);
 	clr_pwr_key();
 	#endif
     
diff --git a/sound/soc/aml/m6/aml_pcm.c b/sound/soc/aml/m6/aml_pcm.c
index 9b67d51b264e..bd38f1d9e6b7 100755
--- a/sound/soc/aml/m6/aml_pcm.c
+++ b/sound/soc/aml/m6/aml_pcm.c
@@ -1000,6 +1000,7 @@ static int aml_pcm_open(struct snd_pcm_substream *substream)
 	int ret = 0;
 	void *buffer = NULL;
 	unsigned int buffersize = 0;
+	audio_stream_t *s = NULL;
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK){
 		playback_substream_handle = (unsigned long)substream;
 		snd_soc_set_runtime_hwparams(substream, &aml_pcm_hardware);
@@ -1059,7 +1060,7 @@ static int aml_pcm_open(struct snd_pcm_substream *substream)
 	
 
 	spin_lock_init(&prtd->s.lock);
-	audio_stream_t *s = &prtd->s;
+	s = &prtd->s;
 	s->xrun_num = 0;
  out:
 	return ret;
diff --git a/sound/soc/aml/m6/aml_platform.c b/sound/soc/aml/m6/aml_platform.c
index eaef49c58909..ca763653b443 100644
--- a/sound/soc/aml/m6/aml_platform.c
+++ b/sound/soc/aml/m6/aml_platform.c
@@ -443,19 +443,19 @@ static struct snd_pcm_ops aml_platform_ops = {
 
 static int aml_platform_pcm_new(struct snd_soc_pcm_runtime *rtd)
 {
-    printk("rtd %x \n",	(unsigned)rtd);
 	
     struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-    printk("cpu_dai %x \n",	(unsigned)cpu_dai);
 	
     struct snd_soc_dai *codec_dai = rtd->codec_dai;
-    printk("codec_dai %x \n",	(unsigned)codec_dai);
 	
     struct aml_audio_interface *aud_interface = NULL;
 //    struct snd_pcm *pcm =rtd->pcm ;
 //    struct snd_soc_card *card = rtd->card;
 	
 	int ret = 0;
+    printk("rtd %x \n",	(unsigned)rtd);
+    printk("cpu_dai %x \n",	(unsigned)cpu_dai);
+    printk("codec_dai %x \n",	(unsigned)codec_dai);
 
     printk(KERN_DEBUG"enter %s (codec_dai: %s %d cpu_dai: %s %d)\n",
                     __FUNCTION__, codec_dai->name, codec_dai->id, cpu_dai->name, cpu_dai->id);
diff --git a/sound/soc/aml/m8/aml_i2s.c b/sound/soc/aml/m8/aml_i2s.c
index c32caa0d9df6..71e87d4aeb7c 100755
--- a/sound/soc/aml/m8/aml_i2s.c
+++ b/sound/soc/aml/m8/aml_i2s.c
@@ -488,8 +488,8 @@ static int aml_i2s_open(struct snd_pcm_substream *substream)
 	struct snd_dma_buffer *buf = &substream->dma_buffer;
 	audio_stream_t *s= &prtd->s;
 	int ret = 0;
-	ALSA_TRACE();
 	unsigned int size = 0;
+	ALSA_TRACE();
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK){
 		snd_soc_set_runtime_hwparams(substream, &aml_i2s_hardware);
 		if(s->device_type == AML_AUDIO_I2SOUT){
diff --git a/sound/soc/aml/m8/aml_m8.c b/sound/soc/aml/m8/aml_m8.c
index 200b9c6d0c87..6e9d61fee5b3 100755
--- a/sound/soc/aml/m8/aml_m8.c
+++ b/sound/soc/aml/m8/aml_m8.c
@@ -89,9 +89,9 @@ static void aml_audio_stop_timer(struct aml_audio_private_data *p_aml_audio)
 static int hp_det_adc_value(struct aml_audio_private_data *p_aml_audio)
 {
     int ret,hp_value,hp_val_sum,loop_num;
+    unsigned int mic_ret = 0;
     hp_val_sum = 0;
     loop_num = 0;
-    unsigned int mic_ret = 0;
     
     while(loop_num < 8){
         hp_value = get_adc_sample(p_aml_audio->hp_adc_ch);
@@ -491,6 +491,7 @@ static int aml_resume_pre(struct snd_soc_card *card)
 
         p_aml_audio->pin_ctl = devm_pinctrl_get_select(card->dev, "aml_snd_m8");
     }
+    printk(KERN_INFO "enter %s\n", __func__);
     return 0;
 }
 
diff --git a/sound/soc/aml/m8/aml_spdif_codec.c b/sound/soc/aml/m8/aml_spdif_codec.c
index 633b472893d6..d0d44b9ebeb9 100755
--- a/sound/soc/aml/m8/aml_spdif_codec.c
+++ b/sound/soc/aml/m8/aml_spdif_codec.c
@@ -101,11 +101,11 @@ static DEVICE_ATTR(spdif_mute, 0660, spdif_mute_show, spdif_mute_set);
 
 static int spdif_dit_probe(struct platform_device *pdev)
 {
+    int ret = device_create_file(&pdev->dev, &dev_attr_spdif_mute);
 	printk("enter spdif_dit_probe \n");
     spdif_dev = &pdev->dev;
 
     aml_spdif_pinmux_init(&pdev->dev);
-    int ret = device_create_file(&pdev->dev, &dev_attr_spdif_mute);
     if (ret < 0)
 		printk("spdif: failed to add spdif_mute sysfs: %d\n",ret);
 	return snd_soc_register_codec(&pdev->dev, &soc_codec_spdif_dit,
diff --git a/sound/soc/aml/m8/aml_spdif_dai.c b/sound/soc/aml/m8/aml_spdif_dai.c
index cdfb36d8952e..f36c6a4d6b69 100755
--- a/sound/soc/aml/m8/aml_spdif_dai.c
+++ b/sound/soc/aml/m8/aml_spdif_dai.c
@@ -42,10 +42,11 @@ static unsigned last_iec_clock =  -1;
 extern int aout_notifier_call_chain(unsigned long val, void *v);
 //static  unsigned  playback_substream_handle = 0 ;
 extern unsigned int IEC958_mode_codec;
-static int iec958buf[32+16];
+//static int iec958buf[32+16];
 void  aml_spdif_play(void)
 {
     return;
+#if 0
    	 _aiu_958_raw_setting_t set;
    	 _aiu_958_channel_status_t chstat;	 
 	struct snd_pcm_substream substream;
@@ -82,6 +83,7 @@ void  aml_spdif_play(void)
 	audio_hw_958_enable(1);
 
 	
+#endif
 }
 static void  aml_spdif_play_stop(void)
 {
diff --git a/sound/soc/codecs/aml_m8_codec.c b/sound/soc/codecs/aml_m8_codec.c
index 800a68ab87c0..7e72e867692a 100755
--- a/sound/soc/codecs/aml_m8_codec.c
+++ b/sound/soc/codecs/aml_m8_codec.c
@@ -180,10 +180,10 @@ static int aml_m8_codec_hw_params(struct snd_pcm_substream *substream,
 
 static int aml_m8_codec_mute_stream(struct snd_soc_dai *dai, int mute, int stream)
 {
-    printk(KERN_DEBUG "enter:%s, mute=%d, stream=%d \n",__func__,mute,stream);
     struct snd_soc_codec *codec = dai->codec;
 	u16 mute_reg = snd_soc_read(codec, AMLM8_MUTE_2) & 0xfc;
     u16 mic_mute_reg = snd_soc_read(codec, AMLM8_MUTE_0) & 0xfc;
+    printk(KERN_DEBUG "enter:%s, mute=%d, stream=%d \n",__func__,mute,stream);
 
     if(stream == SNDRV_PCM_STREAM_PLAYBACK){
     	if (mute){
diff --git a/sound/soc/codecs/wm8960.c b/sound/soc/codecs/wm8960.c
index a18c8fcb4370..00930951f8e3 100755
--- a/sound/soc/codecs/wm8960.c
+++ b/sound/soc/codecs/wm8960.c
@@ -633,10 +633,10 @@ static int wm8960_hw_params(struct snd_pcm_substream *substream,
 
 static int wm8960_mute_stream(struct snd_soc_dai *dai, int mute, int stream)
 {
-    printk(KERN_DEBUG "enter : %s, mute=%d,stream=%d \n",__func__,mute,stream);
     struct snd_soc_codec *codec = dai->codec;
     u16 mute_reg = snd_soc_read(codec, WM8960_DACCTL1) & 0xfff7;
     u16 mute_reg1 = snd_soc_read(codec, WM8960_LINVOL) & 0xff7f;
+    printk(KERN_DEBUG "enter : %s, mute=%d,stream=%d \n",__func__,mute,stream);
     if(stream == SNDRV_PCM_STREAM_PLAYBACK){
         if (mute){
             snd_soc_write(codec, WM8960_DACCTL1, mute_reg | 0x8);
-- 
2.19.0

